streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ThreadMetrics.java

```diff
@@ -0,0 +1,173 @@
+    private static final String CREATE_TASK_RATE_DESCRIPTION = RATE_DESCRIPTION + CREATE_TASK_DESCRIPTION;
+    private static final String CLOSE_TASK_DESCRIPTION = "closed tasks";
+    private static final String CLOSE_TASK_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + CLOSE_TASK_DESCRIPTION;
+    private static final String CLOSE_TASK_RATE_DESCRIPTION = RATE_DESCRIPTION + CLOSE_TASK_DESCRIPTION;
+    private static final String POLL_DESCRIPTION = "poll calls";
+    private static final String POLL_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + POLL_DESCRIPTION;
+    private static final String POLL_RATE_DESCRIPTION = RATE_DESCRIPTION + POLL_DESCRIPTION;
+    private static final String PROCESS_DESCRIPTION = "process calls";
+    private static final String PROCESS_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + PROCESS_DESCRIPTION;
+    private static final String PROCESS_RATE_DESCRIPTION = RATE_DESCRIPTION + PROCESS_DESCRIPTION;
+    private static final String PUNCTUATE_DESCRIPTION = "punctuate calls";
+    private static final String PUNCTUATE_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + PUNCTUATE_DESCRIPTION;
+    private static final String PUNCTUATE_RATE_DESCRIPTION = RATE_DESCRIPTION + PUNCTUATE_DESCRIPTION;
+    private static final String SKIP_RECORDS_DESCRIPTION = "skipped records";
+    private static final String SKIP_RECORD_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + SKIP_RECORDS_DESCRIPTION;
+    private static final String SKIP_RECORD_RATE_DESCRIPTION = RATE_DESCRIPTION + SKIP_RECORDS_DESCRIPTION;
+    private static final String COMMIT_ALL_TASKS_DESCRIPTION = "commit calls over all tasks";
+    private static final String COMMIT_OVER_TASKS_TOTAL_DESCRIPTION = TOTAL_DESCRIPTION + COMMIT_ALL_TASKS_DESCRIPTION;
+    private static final String COMMIT_OVER_TASKS_RATE_DESCRIPTION = RATE_DESCRIPTION + COMMIT_ALL_TASKS_DESCRIPTION;
+
+    private static final String COMMIT_LATENCY = COMMIT + LATENCY_SUFFIX;
+    private static final String POLL_LATENCY = POLL + LATENCY_SUFFIX;
+    private static final String PROCESS_LATENCY = PROCESS + LATENCY_SUFFIX;
+    private static final String PUNCTUATE_LATENCY = PUNCTUATE + LATENCY_SUFFIX;
+
+    private static final String THREAD_LEVEL_GROUP = "stream-metrics";
+    private static final String TASK_LEVEL_GROUP = "stream-task-metrics";
+
+    public static Sensor createTaskSensor(final StreamsMetricsImpl streamsMetrics) {
+        final Sensor createTaskSensor = streamsMetrics.threadLevelSensor(CREATE_TASK, RecordingLevel.INFO);
+        addInvocationRateAndCount(createTaskSensor,
+                                  THREAD_LEVEL_GROUP,
+                                  streamsMetrics.threadLevelTagMap(),
+                                  CREATE_TASK,
+                                  CREATE_TASK_TOTAL_DESCRIPTION,
+                                  CREATE_TASK_RATE_DESCRIPTION);
+        return createTaskSensor;
+    }
+
+    public static Sensor closeTaskSensor(final StreamsMetricsImpl streamsMetrics) {
```

Comment:
This method creates a sensor and corresponding metrics to record when a task closes. See the other methods in this class for the other built-in thread-level sensors.